src/platform_impl/linux/mod.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright 2022-2022 Tauri Programme within The Commons Conservancy
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-License-Identifier: MIT
+
+mod icon;
+mod menu;
+mod tray;
+
+use std::thread;
+
+pub(crate) use icon::PlatformIcon;
+use tray::Tray;
+
+use crate::{icon::Icon, TrayIconAttributes, TrayIconId};
+
+pub struct TrayIcon {
+    tray_handle: ksni::Handle<Tray>,
+}
+
+impl TrayIcon {
+    pub fn new(id: TrayIconId, attrs: TrayIconAttributes) -> crate::Result<Self> {
+        let icon = attrs.icon.map(|icon| icon.inner.into());
+        let title = attrs.title.unwrap_or_default();
+        let tooltip = attrs.tooltip.unwrap_or_default();
+
+        let menu = attrs
+            .menu
+            .as_ref()
+            .map(|menu| menu.compat_items())
+            .unwrap_or_default();
+
+        let tray_service = ksni::TrayService::new(Tray::new(id, icon, title, tooltip, menu));
+        let tray_handle = tray_service.handle();
+        tray_service.spawn();
+
+        let update_tray_handle = tray_handle.clone();
+        thread::spawn(move || {
```

Comment:
we should store the thread handle on `TrayIcon` so we can close this thread on Drop

Reply:
You're right. The thread is not being stopped. But the problem with a thread handle is, that it can only wait for the thread to be closed. I added `send_menu_shutdown` to muda, so the thread can be stopped. But I'm not waiting for it to be stopped, because if something goes wrong, all it would do is block the application.

Reply:
I don't think reusing the same channel to send shutdown signal would work because if you have two tray icons and you drop one of them, it will send a shutdown signal to all

Reply:
Ok, next try: Set an atomic bool and wake the thread up again, so it can shut itself down.